src/librustc_mir/transform/const_prop.rs

```diff
@@ -580,8 +580,20 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
 
                 return None;
             }
+            Rvalue::ThreadLocalRef(def_id) => {
+                trace!("skipping ThreadLocalRef({:?})", def_id);
 
-            _ => {}
+                return None;
+            }
+
+            Rvalue::Aggregate(..)
+            | Rvalue::Use(..)
+            | Rvalue::Repeat(..)
+            | Rvalue::Len(..)
+            | Rvalue::Cast(..)
+            | Rvalue::Discriminant(..)
+            | Rvalue::AddressOf(..)
+            | Rvalue::NullaryOp(..) => {}
```

Comment:
What is the justification for doing nothing for all of these?

I don't actually understand this code, maybe it would be better if @oli-obk reviewed this.^^

Reply:
Would be good to have a brief comment here saying why it is okay to ignore these.